pinot-common/src/main/java/org/apache/pinot/common/utils/config/TableConfigUtils.java

```diff
@@ -212,4 +215,54 @@ public static ZNRecord toZNRecord(TableConfig tableConfig)
     znRecord.setSimpleFields(simpleFields);
     return znRecord;
   }
+
+  /**
+   * Helper method to convert from indexingConfig (that is deprecated) to ingestionConfig.
+   * Since we are moving from indexingConfig to ingestionConfig for ingestion related configs,
+   * this utility helps with that migration. It also handles corner cases where both these
+   * configs might exist, in which case ingestionConfig takes precedence, and only required
+   * fields absent there are carried over from indexingConfig.
+   *
+   * The conversion happens in-place, the specified tableConfig is mutated in-place.
+   *
+   * @param tableConfig Input table config.
+   */
+  public static void convertFromIndexToIngestionConfig(TableConfig tableConfig) {
+    IndexingConfig indexingConfig = tableConfig.getIndexingConfig();
+
+    // It is possible that indexing as well as ingestion configs exist, in which case we always honor ingestion config.
+    IngestionConfig ingestionConfig = tableConfig.getIngestionConfig();
+    BatchIngestionConfig batchIngestionConfig =
+        (ingestionConfig != null) ? ingestionConfig.getBatchIngestionConfig() : null;
+
+    SegmentsValidationAndRetentionConfig validationConfig = tableConfig.getValidationConfig();
+    if (batchIngestionConfig == null) {
+      batchIngestionConfig = new BatchIngestionConfig(null, validationConfig.getSegmentPushType(),
+          validationConfig.getSegmentPushFrequency());
+    } else {
+      // This should not happen typically, but since we are in repair mode, might as well cover this corner case.
+      if (batchIngestionConfig.getSegmentIngestionType() == null) {
+        batchIngestionConfig.setSegmentIngestionType(validationConfig.getSegmentPushType());
+      }
+
+      if (batchIngestionConfig.getSegmentIngestionFrequency() == null) {
+        batchIngestionConfig.setSegmentIngestionFrequency(validationConfig.getSegmentPushFrequency());
+      }
+    }
+
+    StreamIngestionConfig streamIngestionConfig =
+        (ingestionConfig != null) ? ingestionConfig.getStreamIngestionConfig() : null;
+    if (streamIngestionConfig == null) {
+      streamIngestionConfig = new StreamIngestionConfig(Collections.singletonList(indexingConfig.getStreamConfigs()));
+    }
+
+    if (ingestionConfig == null) {
+      ingestionConfig = new IngestionConfig(batchIngestionConfig, streamIngestionConfig, null, null, null);
+    } else {
+      ingestionConfig.setBatchIngestionConfig(batchIngestionConfig);
+      ingestionConfig.setStreamIngestionConfig(streamIngestionConfig);
+    }
+
+    tableConfig.setIngestionConfig(ingestionConfig);
```

Comment:
We should also remove the deprecated configs if exist

Reply:
I haven't checked if there are any callers that rely on the deprecated configs. We can perform the cleanup separately.

Reply:
The callers should use `IngestionConfigUtils` to read the config. Let's at least add a TODO here to do the cleanup.
We should clean up the old configs because I believe this PR is for the auto-tuning feature which will modify certain configs. If we don't clean up the old configs, there will be inconsistency between old and new config, which will cause confusion.

Reply:
Added TODO: https://github.com/apache/incubator-pinot/pull/7050